.gitignore

```diff
@@ -9,6 +9,7 @@
 !/docs
 !/libobs*
 !/plugins
+!/shared
```

Comment:
@RytoEX The entire PR kinda hinges on us agreeing on a directory name for all the modules/headers/elements shared between libobs/plugins/UI, etc.

I'm indifferent to using "shared" but this should be agreed on quick.

Reply:
Acknowledging. Let's discuss off-thread.